py/server/deephaven/parquet.py

```diff
@@ -239,22 +244,30 @@ def write(
     max_dictionary_keys: Optional[int] = None,
     max_dictionary_size: Optional[int] = None,
     target_page_size: Optional[int] = None,
-    generate_metadata_files: Optional[bool] = False,
+    generate_metadata_files: Optional[bool] = None,
 ) -> None:
     """ Write a table to a Parquet file.
 
     Args:
         table (Table): the source table
         path (str): the destination file path; the file name should end in a ".parquet" extension. If the path
-            includes non-existing directories they are created. If there is an error, any intermediate directories
+            includes any non-existing directories, they are created. If there is an error, any intermediate directories
             previously created are removed; note this makes this method unsafe for concurrent use
-        col_definitions (Optional[List[Column]]): the column definitions to use, default is None
-        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while writing, default is None
-        compression_codec_name (Optional[str]): the default compression codec to use, if not specified, defaults to SNAPPY
-        max_dictionary_keys (Optional[int]): the maximum dictionary keys allowed, if not specified, defaults to 2^20 (1,048,576)
-        max_dictionary_size (Optional[int]): the maximum dictionary size (in bytes) allowed, defaults to 2^20 (1,048,576)
+        col_definitions (Optional[List[Column]]): the column definitions to use for writing, instead of the definitions
+            implied by the table. Default is None, which means use the column definitions implied by the table
+        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while writing particular
+            columns, default is None, which means no specialization for any column
+        compression_codec_name (Optional[str]): the default compression codec to use, if not specified, defaults to SNAPPY.
```

Comment:
Given that the compression codec must come from a discrete set, we should probably type hint as a `Literal`.
https://docs.python.org/3/library/typing.html#typing.Literal
Confirm the view with @jmao-denver

Reply:
I have made the change but I am not too confident if I have made it correctly. Basically, the same codec list is to be passed to three methods. So I declared it as a global variable, but that wouldn't give the users any hint, right? Because the list is actually not visible to users.
So should I make copies at three places (in write, batch_write and write_partitioned)?

Reply:
I am not sure using Literal is a good idea here as the supported compression types are more than 3 and could be expanded in the future. That said, it could be helpful to the users to list all the supported types in the docstring.

Reply:
Okay, I will change it back.